src/proxy/inbound.rs

```diff
@@ -294,128 +282,36 @@ impl Inbound {
         // Extract the host or IP from the authority pseudo-header of the URI
         let hbone_addr: HboneAddress = req
             .uri()
-            .clone()
             .try_into()
-            .map_err(|e: Error| InboundError(e, StatusCode::BAD_REQUEST))?;
+            .map_err(InboundError::build(StatusCode::BAD_REQUEST))?;
```

Comment:
Is a lambda less performant or something? Or is this just readability?

Reply:
Just readability, exact same compilation